src/command.jl

```diff
@@ -169,4 +169,49 @@ key(cmd :: OperatorCommand) = cmd.operator
 key(cmd :: LineOperatorCommand) = cmd.operator
 key(cmd :: InsertCommand) = cmd.c
 
+Base.@kwdef struct ParseValue
+    type::Symbol
+    i::Int=1
+    c::Char=' '
+    s::String=""
+end
+ParseValue(::Nothing) = ParseValue(type=:nothing)
+ParseValue(i::Int) = ParseValue(type=:int, i=i)
+ParseValue(c::Char) = ParseValue(type=:char, c=c)
+ParseValue(s::AbstractString) = ParseValue(type=:string, s=s)
+
+"""
+This function unpacks the parsed values into a `Command` object.
+
+For example, the code:
+```
+ReplaceCommand(1, 'c')
+```
+is equivalent to the code:
+```
+command_constructor(ReplaceCommand, ParseValue(1), ParseValue('c'))
+```
+The reason to use this second approach is better type stability.
+(Though, a refactor with a single type-stable command struct would be better.)
```

Comment:
Does this mean a single concrete struct?

Reply:
A single struct like e.g., `Command(…)` where the fields of the struct indicate what the command does and what its inputs are. (Rather than a type for each command)